script.js

```diff
@@ -1,12 +1,602 @@
+    let showImg = show && show.image ? show.image.medium : `https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRqEWgS0uxxEYJ0PsOb2OgwyWvC0Gjp8NUdPw&usqp=CAU`;
+    return `
+        <div class="card shows-genre-card">
+          <div class="col-10">
+            <img src="${showImg}" class="card-img-top " alt="...">
+          </div>
+          <div class="card-body p-0">
+            <h5 id="${show.id}" class="card-title show-name">${show.name}</h5>
+            <p class="card-text">${show.rating.average}</p>
+          </div>
+        </div>
+        `
+  }).join('');
+  topRatedShows.innerHTML = topRatedShowsHtml;
+  
+  const allShowGenres = getGenreListing();
+  const allDramaShows = allShowGenres[0].Drama;
+  const allComedyShows = allShowGenres[0].Comedy;
+  const allCrimeShows = allShowGenres[0].Crime;
+  const allActionShows = allShowGenres[0].Action;
+  const allScifiShows = allShowGenres[0]['Science-Fiction'];
+
+  let dramaShowsHtml = allDramaShows.slice([10],[20]).map((show) => { 
+    let showImg = show && show.image ? show.image.medium : `https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRqEWgS0uxxEYJ0PsOb2OgwyWvC0Gjp8NUdPw&usqp=CAU`;
+    return `
+        <div class="card shows-genre-card">
+          <div class="col-10">
+            <img src="${showImg}" class="card-img-top " alt="...">
+          </div>
+          <div class="card-body p-0">
+            <h5 id="${show.id}" class="card-title show-name">${show.name}</h5>
+            <p class="card-text">${show.rating.average}</p>
+          </div>
+        </div>
+        `
+  }).join('');
+  dramaShows.innerHTML = dramaShowsHtml;
+  let comedyShowsHtml = allComedyShows.slice([0],[10]).map((show) => { 
+    let showImg = show && show.image ? show.image.medium : `https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRqEWgS0uxxEYJ0PsOb2OgwyWvC0Gjp8NUdPw&usqp=CAU`;
+    return `
+        <div class="card shows-genre-card">
+          <div class="col-10">
+            <img src="${showImg}" class="card-img-top " alt="...">
+          </div>
+          <div class="card-body p-0">
+            <h5 id="${show.id}" class="card-title show-name">${show.name}</h5>
+            <p class="card-text">${show.rating.average}</p>
+          </div>
+        </div>
+        `
+  }).join('');
+  comedyShows.innerHTML = comedyShowsHtml;
+  let crimeShowsHtml = allCrimeShows.slice([0],[10]).map((show) => { 
+    let showImg = show && show.image ? show.image.medium : `https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRqEWgS0uxxEYJ0PsOb2OgwyWvC0Gjp8NUdPw&usqp=CAU`;
+    return `
+        <div class="card shows-genre-card">
+          <div class="col-10">
+            <img src="${showImg}" class="card-img-top " alt="...">
+          </div>
+          <div class="card-body p-0">
+            <h5 id="${show.id}" class="card-title show-name">${show.name}</h5>
+            <p class="card-text">${show.rating.average}</p>
+          </div>
+        </div>
+        `
```

Comment:
I am just wondering whether we could create a `function()` to return a related output by passing the related parameter. Because all codes below are repeated 5 times. I am just curious to make the code a bit shorter by using one reusable function. You might have tried this. If it is not achievable and functionable, that's fine.

```
  let comedyShowsHtml = allComedyShows.slice([0],[10]).map((show) => { 
    let showImg = show && show.image ? show.image.medium : `https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRqEWgS0uxxEYJ0PsOb2OgwyWvC0Gjp8NUdPw&usqp=CAU`;
    return `
        <div class="card shows-genre-card">
          <div class="col-10">
            <img src="${showImg}" class="card-img-top " alt="...">
          </div>
          <div class="card-body p-0">
            <h5 id="${show.id}" class="card-title show-name">${show.name}</h5>
            <p class="card-text">${show.rating.average}</p>
          </div>
        </div>
        `
  }).join('');
```

Reply:
Yep you are right, neatening up my code and and made it so 'DRY'